src/main/scala/org/hammerlab/guacamole/readsets/iterator/ContigCoverageIterator.scala

```diff
@@ -53,11 +52,28 @@ case class ContigCoverageIterator(halfWindowSize: Int,
       else
         lastStart = start
 
-      ends.enqueue(end)
-      numAdded += 1
+      if (end + halfWindowSize > locus) {
```

Comment:
yea, good idea, added a bunch!

also while explaining myself, found an edge case that I still wasn't handling correctly here, so fixed and added a test for it. thanks for prompting!